docs: Move in production with bonita-ui-builder (#2739)
bonitasoft/bonita-doc: modules/applications/pages/package-and-deploy-your-application.adoc

= Package and deploy your application
:description: Step-by-step guide to transition from development to production using Bonita UI Builder

[NOTE]
====
For Subscription editions only.
====

Once you've designed an application that interacts with your xref:interact-with-your-bonita-process.adoc[bonita process], and xref:builder-declare-interface-in-bonita.adoc[declared] it in Bonita, it's time to push your interface to production

== Prerequisites

> Before starting, ensure you have **Docker** the following tools installed on your system.

Reviewer: This prerequisite is for all the UI Builder usage, so I suggest to not put it there. If you do, remove "the following tools".

== Move from Development to Production

=== Export Your Application

> Once the design is complete, export the application(s) from Bonita UI Builder by clicking the `export` button on the home page. This export will be used for building the production binary.

Reviewer: Give more details on where the "export" button is located, this is not obvious...

=== Store Your Application in a Workspace Folder

> Place the exported application(s) in a designated `workspace` folder. This folder will be referenced during the build process.

Reviewer: ```suggestion
Move the exported application(s) (json files) in a designated `workspace` folder. This folder will be referenced during the build process.
```

You can store these applications in a Git repository too.

=== Build the Production Binary

To create the production-ready binary, you need to build a Docker image. This image will contain the application(s) and the Bonita UI Builder runtime.

The xref:applications:production-packaging.adoc#dockerfile[Dockerfile] and more details and information on how to build the Docker image can be found in this xref:applications:production-packaging.adoc[dedicated page].

=== Run the Docker Image

After building the Docker image, run it using the following command:

[source,shell]
----
docker run --name <container-name> <image-name>:<version>
----
See xref:production-packaging.adoc#run-production-binary[run production binary page] for more details.

=== Access to your Application

Once Bonita UI Builder is ready, all applications contained in the dedicated folder will be:

- Imported
- Published
- Made available for users

> Then, these applications can be accessed from the Bonita application directory

Reviewer: ```suggestion
Then, these applications can be accessed from the Bonita administrator application, application directory page.
```

[.troubleshooting-title]
== image:ROOT:images/troubleshooting.png[troubleshooting-icon] Troubleshooting

[.troubleshooting-section]
--
[.symptom]
After building and running the Bonita UI Builder, at least one application are missing.

[.cause]#Cause#
The application is on error and the Bonita UI Builder is not able to import and deploy the application.

[.solution]#Solution#

> Check the log of the bonita-ui-builder container with `docker logs <container-name>` to find with application is on error.

Reviewer: ```suggestion
Check the log of the bonita-ui-builder container with `docker logs <container-name>` to find which application is on error.
```
